x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -373,11 +405,31 @@ public InferenceProcessor create(
                 inferenceConfigUpdate = inferenceConfigUpdateFromMap(inferenceConfigMap);
             }
 
-            List<Map<String, Object>> inputs = ConfigurationUtils.readOptionalList(TYPE, tag, config, INPUT_OUTPUT);
+            Object inputOutputs = config.remove(INPUT_OUTPUT);
+            List<Map<String, Object>> inputs = null;
+            if (inputOutputs != null) {
```

Comment:
Might be worth moving this to a separate method since there's a bunch of if-block nesting, that might make reading `create()` a little easier 🤷‍♂️